codegen/codegen.go

```diff
@@ -1021,6 +1026,11 @@ func (cg *CodeGen) EmitExecOptions(ctx context.Context, scope *parser.Scope, op
 					return opts, err
 				}
 
+				localPath, err = ResolvePathForNode(scope.Node, localPath)
+				if err != nil {
+					return opts, err
+				}
```

Comment:
do we need to add SharedKeyHint for secrets as well?

Reply:
It's only a local option atm:
https://github.com/moby/buildkit/blob/4ae4bcd6a08e3a40aedead8d2edd571a11f75b55/client/llb/source.go#L345-L349